src/bitdrift_public/protobuf/workflow/v1/workflow.proto

```diff
@@ -76,6 +79,53 @@ message Workflow {
     }
   }
 
+  // An extra configuration for a given transition.
+  message TransitionExtension {
+    oneof annotation_type {
+      option (validate.required) = true;
+      SankeyDiagramValueExtraction sankey_diagram_value_extraction = 1;
+    }
+
+    // The configuration of a value extraction for a Sankey diagram.
+    message SankeyDiagramValueExtraction {
+      // The ID of the Sankey diagram to extract the value for.
+      string sankey_diagram_id = 1 [(validate.rules).string = {min_len: 1}];
+
+      // The maximum number of values that can be extracted. Values are kept in FIFO
+      // order, and as soon as the limit is exceeded, the oldest ones are dropped to make space for new
+      // ones. A limit of 0 means this feature is effectively disabled, and all states for a given
+      // transition are retained.
+      uint32 limit = 2;
+
+      // The name of the Sankey diagram state.
+      oneof value_type {
+        option (validate.required) = true;
+
+        // A fixed value.
+        string fixed = 3 [(validate.rules).string = {min_len: 1}];
+
+        // The value extracted from a field.
+        FieldExtracted field_extracted = 4;
+      }
+
+      // Extracts the value from a field.
+      message FieldExtracted {
```

Comment:
I think we have this duplicated in a few places, wonder if we should reuse? Not a blocker, just curious